File: python/src/checking_account.py
from src.account import Account


class CheckingAccount(Account):
    def __init__(self, account_holder, balance, account_number, overdraft_protection):
        super().__init__(account_holder, balance, account_number)
        # TODO: Implement constructor

    def get_overdraft_protection(self):
        # TODO: Implement getter
        return False

    def set_overdraft_protection(self, overdraft_protection):
        # TODO: Implement setter
        pass

    def debit(self, amount):
        # TODO: Implement debit method
        # If overdraft_protection is True, don't allow balance to go negative
        # If overdraft_protection is False, allow balance to go negative
        pass

File: src/main/java/io/zipcoder/CheckingAccount.java
package io.zipcoder;

public class CheckingAccount extends Account {
    private boolean overdraftProtection;

    public CheckingAccount(Object accountHolder, Double balance, String accountNumber, boolean overdraftProtection) {
        super(accountHolder, balance, accountNumber);
        // TODO: Implement constructor
        this.overdraftProtection = overdraftProtection;
    }

    public boolean getOverdraftProtection() {
        // TODO: Implement getter
        return overdraftProtection;
    }

    public void setOverdraftProtection(boolean overdraftProtection) {
        // TODO: Implement setter
        this.overdraftProtection = overdraftProtection;
    }

    @Override
    public void debit(Double amount) {
        // TODO: Implement debit method
        // If overdraftProtection is true, don't allow balance to go negative
        // If overdraftProtection is false, allow balance to go negative
        if(this.overdraftProtection && amount >= getBalance()alance()) {
            System.out.println("Transaction Declined");
            return;            System.out.println("Transaction Declined");
         
        double newBalance = getBalance() - amount;   return;
        }
        double newBalance = getBalance() - amount;
    }
}
